### PR TITLE
database migrations/store methods for document collections

### PR DESCRIPTION
**Pull request by n1ru4l**

### CODE REVIEW

File: integration-tests/package.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
   "private": true,
   "scripts": {
     "prepare:env": "cd ../ && pnpm build:libraries && pnpm build:services",
-    "test:integration": "vitest ."
+    "test:integration": "vitest"
```
Comment (n1ru4l) [inline integration-tests/package.json:8]: removed this so i can run tests with filter locally. (`pnpm --filter integration-tests test:integration document-collection`)

File: packages/services/cdn-worker/package.json [hunk 2]
```diff
@@ -3,6 +3,11 @@
   "version": "0.0.0",
   "license": "MIT",
   "private": true,
+  "exports": {
+    "./aws": "./src/aws.ts",
+    "./cdn-token": "./src/cdn-token.ts",
+    "./artifact-storage-reader": "./src/artifact-storage-reader.ts"
+  },
```
Comment (n1ru4l) [inline packages/services/cdn-worker/package.json:10]: Had to add these for vitest to not complain

File: packages/services/storage/package.json [hunk 3]
```diff
@@ -7,6 +7,7 @@
   "engines": {
     "node": ">=12"
   },
+  "main": "./src/index.ts",
```
Comment (n1ru4l) [inline packages/services/storage/package.json:10]: Had to add these for vitest to not complain

File: integration-tests/tests/storage/document-collection.spec.ts [hunk 4]
```diff
@@ -0,0 +1,487 @@
+/* eslint-disable no-process-env */
+// eslint-disable-next-line import/no-extraneous-dependencies
+import 'reflect-metadata';
```
Comment (n1ru4l) [inline integration-tests/tests/storage/document-collection.spec.ts:3]: without this vitest complains

File: integration-tests/tests/storage/document-collection.spec.ts [hunk 5]
```diff
@@ -0,0 +1,487 @@
...
+import { ProjectType } from 'testkit/gql/graphql';
+import { initSeed } from 'testkit/seed';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { createStorage } from '@hive/storage';
```
Comment (n1ru4l) [inline integration-tests/tests/storage/document-collection.spec.ts:7]: if i add it to dependencies, it complains. if I add it to devDependencies it complains. So I ignored this.

File: integration-tests/tests/storage/document-collection.spec.ts [hunk 6]
```diff
@@ -0,0 +1,487 @@
...
+import { crypto } from '@whatwg-node/fetch';
+
+const connectionString = `pg://${process.env.POSTGRES_USER!}:${process.env
+  .POSTGRES_PASSWORD!}@localhost:5432/${process.env.POSTGRES_DB!}`;
+
+const makeStorage = () => {
+  return createStorage(connectionString, 1);
+};
+
+describe('document collection', () => {
```
Comment (n1ru4l) [inline integration-tests/tests/storage/document-collection.spec.ts:17]: Added these tests to verify that db operations work as expected. Feel free to convert this to GraphQL API tests (similar to https://github.com/kamilkisiela/graphql-hive/blob/03359b8093bd4683740b739ea61cc610ff231997/integration-tests/tests/api/artifacts-cdn.spec.ts#L616)

File: packages/services/storage/src/index.ts [hunk 7]
```diff
@@ -2995,16 +2995,322 @@ export async function createStorage(connection: string, maximumPoolSize: number)
...
+      `);
+
+      if (result == null) {
+        return null;
+      }
+
+      return zod.string().parse(result);
+    },
+
+    async updateDocumentCollectionDocument(args) {
```
Comment (B2o5T) [inline packages/services/storage/src/index.ts:3274]: I think we should rethink name for collection

File: packages/services/storage/src/index.ts [hunk 8]
```diff
@@ -2995,16 +2995,322 @@ export async function createStorage(connection: string, maximumPoolSize: number)
...
+      const result = await pool.maybeOne(sql`
+        UPDATE
+          "public"."document_collection_documents"
+        SET
+          "title" = COALESCE(${args.title}, "title")
```
Comment (B2o5T) [inline packages/services/storage/src/index.ts:3279]:
```suggestion
          "name" = COALESCE(${args.title}, "name")
```
I want rename this column to name

File: packages/services/storage/src/index.ts [hunk 9]
```diff
@@ -2995,16 +2995,322 @@ export async function createStorage(connection: string, maximumPoolSize: number)
...
+          , "contents" = COALESCE(${args.contents}, "contents")
```
Comment (B2o5T) [inline packages/services/storage/src/index.ts:3280]:
```suggestion
          , "query" = COALESCE(${args.contents}, "query")
```
I want rename this column to `query`

**Comment by B2o5T** [comment]
@n1ru4l idk how I merged it, I just pushed new changes to my branch `share-query` lol

**Pull request merged**